test/tests/functional/pbs_basil_support.py

```diff
@@ -308,3 +309,42 @@ def test_cray_login_node(self):
                              ("%s\t: pbs node has %s=%s but login "
                               "node has %s=%s") %
                              (mom_id, rsc, val, rsc, rval))
+
+    def test_hbmemm_rsc(self):
+        """
+        Create a job that requests enough HBMEM. Submit the job to
+        the Server. Check if the job is in the 'R' state and if the
+        job runs on a KNL vnode. Delete the job.
+        """
+
+        knl_vnodes = self.get_knl_vnodes()
+
+        if len(knl_vnodes) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (knl_vnodes))
+
+        # Add 'hbmem' to the resources line in sched_config.
+        self.scheduler.add_resource('hbmem')
+
+        a = {'Resource_List.select': '1:hbmem=4192'}
+        job = Job(TEST_USER, attrs=a)
+
+        job_id = self.server.submit(job)
+        self.server.expect(JOB, {'job_state': 'R'}, id=job_id)
+
+        # Check that exec_vnode is a KNL vnode.`
+        self.server.status(JOB, 'exec_vnode', id=job_id)
+        evnode = job.execvnode()[0].keys()[0]
+        nid = evnode.split('_')[1]
+        if nid in knl_vnodes.keys():
+            self.logger.info("exec_vnode %s is a KNL vnode." % (evnode))
+            rv = 1
+        else:
+            self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
+            rv = 0
+        self.assertTrue(rv == 1)
+
```

Comment:
Since the resource is consumable, shouldn't you also check that the amount requested is what got assigned to the job?

Reply:
Done, Added check to verify requested hbmem is assigned to the job.